templates/master/00-master/azure/units/openshift-azure-routes.path.yaml

```diff
@@ -0,0 +1,17 @@
+name: openshift-azure-routes.path
+enabled: true
+contents: |
+  [Unit]
+  Description=Watch for downfile changes
+  Wants=network-online.target
+  After=network-online.target
+  Before=kubelet.service crio.service
```

Comment:
Path units get started with `paths.target` by default. So this is creating an ordering cycle because `network-online.target` wants to happen after `basic.target`:

```
[  324.981226] systemd[1]: network-online.target: Found ordering cycle on network.target/start
[  325.000115] systemd[1]: network-online.target: Found dependency on ovs-vswitchd.service/start
[  325.013025] systemd[1]: network-online.target: Found dependency on basic.target/start
[  325.027061] systemd[1]: network-online.target: Found dependency on paths.target/start
[  325.035592] systemd[1]: network-online.target: Found dependency on openshift-azure-routes.path/start
[  325.049473] systemd[1]: network-online.target: Found dependency on network-online.target/start
[  325.058497] systemd[1]: network-online.target: Job network.target/start deleted to break ordering cycle starting with network-online.target/start
[ SKIP ] Ordering cycle found, skipping Network
```

(this is from the serial console logs from https://bugzilla.redhat.com/show_bug.cgi?id=1877483)

Depending on which unit systemd decides to delete, networking might not come up.

Hmm, I think we probably need to add `DefaultDependencies=false` here?

Reply:
Also related to this, we should patch the MCO to add an alert if it detects a systemd ordering cycle.

Reply:
Looking at this I think those dependencies are just wrong - we will only write files in there when the static pod comes up, so kubelet will be up for that.  IOW I think we want:

```
diff --git a/templates/master/00-master/azure/units/openshift-azure-routes.path.yaml b/templates/master/00-master/azure/units/openshift-azure-routes.path.yaml
index 75298150..a3795a30 100644
--- a/templates/master/00-master/azure/units/openshift-azure-routes.path.yaml
+++ b/templates/master/00-master/azure/units/openshift-azure-routes.path.yaml
@@ -3,9 +3,6 @@ enabled: true
 contents: |
   [Unit]
   Description=Watch for downfile changes
-  Wants=network-online.target
-  After=network-online.target
-  Before=kubelet.service crio.service
   ConditionPathExists=!/etc/ignition-machine-config-encapsulated.json
 
   [Path]
@@ -14,4 +11,4 @@ contents: |
   MakeDirectory=true
 
   [Install]
-  WantedBy=multi-user.target
+  WantedBy=paths.target
```

Reply:
Force pushed an update similar to that.

Reply:
> Also related to this, we should patch the MCO to add an alert if it detects a systemd ordering cycle.

https://github.com/openshift/machine-config-operator/issues/2074